File: Classes/RandomInteger.m
//
//  RandomInteger.m
//  MathFlashCards
//
//  Created by Jagmeet Chawla on 4/12/09.
//  Revised by Krishna Narayan on 9/17/09
//		--Updated graphics
//		--Added tell a friend
//		--Changed app name to Kids Learn To Tell Time
//  Copyright 2009-2013 NSC Partners LLC. Copyright 2014 One Step Ahead Apps, LLC. All rights reserved.
// Revised by Krishna Narayan on 5/30/26 — Used Claude to migrate to Swift, fix UI Views, remove deprecations, update for iPad, modernize for Apple UI rules.
// Copyright 2026 Island Innovation LLC.  All rights reserved.
//

#import "RandomInteger.h"

@implementation RandomInteger

@synthesize rangeLow;
@synthesize rangeHigh;
@synthesize randomInteger;

- (id) init {
	if (self = [super init]) {
		srandom((unsigned int)time(NULL));
	}
	return self;
}

- (id) initWithRange: (int)low To: (int)high {
//	[self init];    //satyam
    if (self = [super init])
    {
        self.rangeLow = low;
        self.rangeHigh = high;
    }
	return self;
}
		
- (int) nextRandomInteger {
	int r;
	r = random() % (self.rangeHigh-self.rangeLow+1) + self.rangeLow; //generate a random number in given range
	srandom((unsigned int)time(NULL)*(unsigned int)random()); //seed again, this time with a random number
	return r;
}

- (int) nextRandomIntegerInRange: (int)low To: (int)high {
	self.rangeLow = low;
	self.rangeHigh = high;
	return [self nextRandomInteger];
}

@end

File: Classes/RandomInteger.swift
// Revised by Krishna Narayan on 5/30/26 — Used Claude to migrate to Swift, fix UI Views, remove deprecations, update for iPad, modernize for Apple UI rules.
// Revised by Krishna Narayan on 6/3/26 — Using Claude changed to 1st, 2nd, and 3rd grade levels, belts are earned not selected, added adaptive weak-drilling algorithm to rectify mistakes and build proficiency after initially providing randomized problems for activities
// Copyright 2026 Island Innovation LLC.  All rights reserved.

import Foundation

class RandomInteger: NSObject {

    var rangeLow: Int
    var rangeHigh: Int

    var randomInteger: Int {
        return nextRandomInteger()
    }

    init(range low: Int, to high: Int) {
        rangeLow = low
        rangeHigh = high
    }

    func nextRandomInteger() -> Int {
        return Int.random(in: rangeLow...rangeHigh)
    }

    func nextRandomInteger(inRange low: Int, to high: Int) -> Int {
        rangeLow = low
        rangeHigh = high
        return nextRandomInteger()
    }
}
